.github/workflows/release.yaml

```
@@ -1,83 +1,131 @@
+# It's triggered once a release PR (created by the pre-release workflow) gets merged to the `main` branch.
 name: Release
-
 on:
-  workflow_dispatch:
-    inputs:
-      bumpType:
-        description: Semver bump type to use ("major"/"minor"/"patch")
-        required: true
-        default: patch
+  # Once the PR gets merged to `main`
+  pull_request:
+    branches:
+      - main
+    types: [closed]
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
+  # Enable release process either with workflow dispatch or the automated PR process
   release:
-    strategy:
-      matrix:
-        os:
-          - ubuntu-latest
-    runs-on: ${{ matrix.os }}
+    name: Release
+    if: |
+      ((github.event.pull_request.merged == true) &&
+      startsWith(github.event.pull_request.title, 'chore(release)'))
+    runs-on: blacksmith-2vcpu-ubuntu-2204
     steps:
-      - name: Checkout code
+      - name: Checkout Repo
         uses: actions/checkout@v4
         with:
-          fetch-depth: '0'
-          submodules: recursive
-
-      - id: release
-        uses: anothrNick/github-tag-action@1.39.0
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DEFAULT_BUMP: ${{ github.event.inputs.bumpType }}
-          TAG_CONTEXT: branch
-          RELEASE_BRANCHES: main,releases.*
-          WITH_V: true
-          DRY_RUN: true
-
-      - id: parsed-release
-        uses: booxmedialtd/ws-action-parse-semver@v1
+          persist-credentials: false
+      - name: Setup tools
+        uses: ./.github/actions/setup-tools
+        with:
+          install-nodejs: 'true'
+      # Publishes a release in case the release isn't published
+      - name: Publish release
+        id: publish-release
+        uses: changesets/action@v1
         with:
-          input_string: ${{ steps.release.outputs.tag }}
+          publish: npm run release
+          createGithubReleases: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      # Prepare the variables that will be used across the different next steps
+      - name: Prepare cross-steps variables
+        run: |
+          export PACKAGE_NAME=$(jq -r '.name' 'package.json');
+          export PUBLISHED_PACKAGE_VERSION=v$(echo '${{ steps.publish-release.outputs.publishedPackages }}' | jq -r ".[] | select(.name == \"${PACKAGE_NAME}\") | .version")
+          export BASE_ARTIFACTS_SEARCH_DIR="./artifacts"
+          export ARTIFACT_VERSIONED_PACKAGE_NAME="interchain-token-service-assets-${PUBLISHED_PACKAGE_VERSION}"
+          export BASE_ARTIFACTS_VERSIONED_DIR="$(dirname ${BASE_ARTIFACTS_SEARCH_DIR})/${ARTIFACT_VERSIONED_PACKAGE_NAME}" # Regardless of the dir type, relative or absolute
+
+          # ex: @axelarnetwork/interchain-token-service
+          echo "PACKAGE_NAME=${PACKAGE_NAME}" >> $GITHUB_ENV
```

```suggestion
          echo "package_name=${PACKAGE_NAME}" >> $GITHUB_OUTPUT
```
use github outputs with lower case names so it's a bit more readable later maybe? too many full cap vars

---

i'm also ok with making the env var names more concise instead